Dockerfile

```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
 chmod -R +r /opt/pseudos/
 
 # Install Python packages needed for AiiDA lab.
-RUN pip install                    \
-    'aiidalab==v20.05.0b1'         \
+RUN pip install 'aiidalab==v20.05.0b1'
+
+# Installing Jupyter-related things in the root environment.
+RUN /usr/bin/pip3 install          \
     'jupyterhub==0.9.4'            \
     'jupyterlab==0.35.4'           \
-    'nbserverproxy==0.8.8'
-
-# Enable nbserverproxy extension.
-RUN jupyter serverextension enable --sys-prefix --py nbserverproxy
+    'fileupload==0.1.5'            \
+    'nbserverproxy==0.8.8'         \
+    'appmode-aiidalab==0.5.0.1'    \
+    'notebook==5.7.8'              \
+    'nglview'                      \
+    'voila'
+
+# Enable extensions.
+RUN /usr/local/bin/jupyter serverextension enable --sys-prefix --py nbserverproxy
+RUN /usr/local/bin/jupyter nbextension     enable --py --sys-prefix appmode
+RUN /usr/local/bin/jupyter serverextension enable --py --sys-prefix appmode
+RUN /usr/local/bin/jupyter nbextension enable nglview --py --sys-prefix
+# TODO: delete, when https://github.com/aiidalab/aiidalab-widgets-base/issues/31 is fixed
+# the fileupload extension also needs to be "installed".
+RUN /usr/local/bin/jupyter nbextension install --sys-prefix --py fileupload
 
 # Enables better integration with Jupyter Hub.
 # https://jupyterlab.readthedocs.io/en/stable/user/jupyterhub.html#further-integration
 # Takes about 3 minutes and 20 seconds.
 RUN jupyter labextension install @jupyterlab/hub-extension
 
-# TODO: delete, when https://github.com/aiidalab/aiidalab-widgets-base/issues/31 is fixed
-# the fileupload extension also needs to be "installed".
-RUN jupyter nbextension install --sys-prefix --py fileupload
-
 # Install jupyterlab theme.
```

Comment:
by the way, as long as we aren't using it, I think it's perfectly fine to comment out anything related to the jupyterlab
(not sure whether the latest voilà stuff is starting to use it?)

Reply:
I think Dou might start using it, so let's keep it.